modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java

```diff
@@ -143,15 +135,39 @@ private void _assertPagination(int start, int end, int expectedSize)
 		searchContext.setEnd(end);
 		searchContext.setStart(start);
 
-		Hits hits = indexSearcher.search(searchContext, getDefaultQuery());
+		IdempotentRetryAssert.retryAssert(
```

Comment:
Can we avoid using IdempotentRetryAssert? I really don't like this, by using this it means we have random behavior. There is no way to know 3sec(or any other time) is enough.